sssd_test_framework/utils/tools.py

```diff
@@ -515,6 +515,18 @@ def grep(self, pattern: str, paths: str | list[str], args: list[str] | None = No
 
         return command.rc == 0
 
+    def connected_interface(self) -> str:
+        result = self.host.conn.exec(["ip", "route", "show", "default"])
+        if result.rc == 0:
+            output = result.stdout.strip()
+            if output:
+                interface = output.split()[4]  # The interface name is the 5th field
+                return interface
+            else:
+                return "No default route found."
```

Comment:
This would hide a failure pushing unexpected string to other places.
Either return empty string (or None) to make the failure obvious or raise an exception.